lib/tokenize.cpp

```diff
@@ -8485,6 +8485,10 @@ void Tokenizer::simplifyKeyword()
             // 1) struct name final { };   <- struct is final
             if (Token::Match(tok->previous(), "struct|class|union %type%")) {
                 Token* finalTok = tok->next();
+                if (tok->isUpperCaseName() && Token::Match(finalTok, "%type%") && finalTok->str() != "final") {
```

Comment:
hmm.. this is a guessing game.. theoretically I think your guess could go wrong in such code:

    class C FINAL {};

Could we try to bailout if we are unsure?

You could make this check more explicit. You could check that finalTok is not uppercase. Or how about `Token::Match(finalTok, "%type% final")`

Reply:
`class C FINAL {};` gives `test.cpp:1:1: information: The code 'class C FINAL {' is not handled. You can use -I or --include to add handling of this code. [class_X_Y]` even without this change.
We cannot bail out in the tokenizer, since it might be a valid variable declaration, see https://github.com/danmar/cppcheck/actions/runs/3920131345/jobs/6701567873